contracts/utils/math/Math.sol

```diff
@@ -149,4 +149,62 @@ library Math {
         }
         return result;
     }
+
+    /**
+     * @dev Returns the square root of a number.
+     *
+     * Inspired by Henry S. Warren, Jr.'s "Hacker's Delight" (Chapter 11).
+     */
+    function sqrt(uint256 a) internal pure returns (uint256) {
+        if (a == 0) {
+            return 0;
+        }
+
+        // For our first guess, we use the log2 of the square root. We do that by shifting `a` and only counting half
+        // of the bits. the partial result produced is a power of two that verifies `result <= sqrt(a) < 2 * result`
```

Comment:
I don't think this is clear or accurate.

> For our first guess, we use the log2 of the square root.

What does it mean that we "use the log2"? I read this like saying that our guess is `log2(sqrt(a))`, but I don't think this is true.

I also find the part about shifting `a` confusing.